### ISSUE TITLE
Different snapshot results with Jest `--coverage` flag and `withComponent` usage

### ISSUE THREAD

**Issue by austin94** [styled-components/jest-styled-components#186]
### Description

Not sure if this is something wrong with my understanding of jest-styled-components and the snapshotSerializer, but it appears that any `withComponent` usage in styled-components can return different snapshot values based on the [--coverage](https://jestjs.io/docs/en/cli.html#coverage) flag in Jest.

It appears that in some cases, additional `-TAG` (`-div` in this case) classes are added.

#### Offending Components

Here are some examples of the `withComponent` usage and the Travis run (with failing snapshots) https://github.com/zendeskgarden/react-components/pull/146/files

#### Without `--coverage`

![screen shot 2018-09-10 at 8 49 42 am](https://user-images.githubusercontent.com/4030377/45308508-7b557800-b4d6-11e8-99d0-f5414ad418bd.png)

#### With `--coverage`

![screen shot 2018-09-10 at 8 50 46 am](https://user-images.githubusercontent.com/4030377/45308595-a93abc80-b4d6-11e8-842b-4dfa1246d3ce.png)

Any ideas/thoughts on how this could be happening are appreciated.  Thanks 👋

**Comment by austin94** [comment]
Weird, looks like this is no longer causing an extra class to be added.

For reference:
- [Failing build](https://travis-ci.org/zendeskgarden/react-components/builds/426717953)
- [Passing build](https://travis-ci.org/zendeskgarden/react-components/builds/426771072)

Going to close for now since it doesn't seem to be reproducing. Feel free to open it again if you feel it might affect anyone else.

**Comment by austin94** [comment]
Looks like this is still occurring actually 😢https://travis-ci.org/zendeskgarden/react-components/builds/426830282

Going to re-open

**Comment by MicheleBertoli** [comment]
Hello @austin94, thanks for opening this issue.

Does it repro consistently?
I cloned the repo but tests don't fail when I run:
`yarn jest --config=utils/test/jest.config.js packages/typography --runInBand --coverage`

**Comment by austin94** [comment]
👋 Hi @MicheleBertoli, thanks for taking a look.  We ended up committing the snapshot changes so it's no longer failing.

After some digging, it looked like this might have been caused by a change in styled-components and not this repo. This was only affecting components that were using the `withComponent()` API where the original component didn't have a valid `displayName`.

We also seemed to have an issue in our `yarn.lock` file which was producing different results based on a local install vs. `yarn install --frozen-lockfile` in Travis, this was causing the miss-match. After re-creating the lockfile we had consistent results.

Going to close again. Thanks for taking a look.
